src/usvfs_dll/sharedparameters.cpp

```diff
@@ -198,6 +200,62 @@ bool SharedParameters::executableBlacklisted(
   return false;
 }
 
+void SharedParameters::addSkipFileSuffix(const std::string& fileSuffix)
+{
+  bi::scoped_lock lock(m_mutex);
+
+  m_fileSuffixSkipList.insert(shared::StringT(fileSuffix.begin(), fileSuffix.end(),
+                                              m_fileSuffixSkipList.get_allocator()));
+}
+
+void SharedParameters::clearSkipFileSuffixes()
+{
+  bi::scoped_lock lock(m_mutex);
+  m_fileSuffixSkipList.clear();
+}
+
+bool SharedParameters::fileShouldBeSkipped(const std::string& file) const
+{
+  bi::scoped_lock lock(m_mutex);
```

Comment:
We should be able to use a read write lock, like the shared_mutex, so we can have multiple readers at the same time.

Reply:
That seems like a change for a different PR, this lock uses the same mutex used throughout this class

Reply:
That seems out of scope for this PR, this lock uses a mutex used throughout this class

Reply:
I looked at the code again, so the problem is that we are calling those functions from a hot path, on every file that we add to the Redirect tree, which can be hundred of thousands of files. 

Every time we lock everything because we are accessing the shared parameters.
This isn't great, we should instead get the excluded lists once from shared memory, and then reuse that for all files, so we can bypass the lock. We don't expect the blacklist to be updated in the middle of the files being added, in if it was, we can ignore the changes to it since we already did a number of files with the previous filter anyways.

Do you think this proposed change is more doable?

Reply:
That makes complete sense.

Instead of `file/directory..ShouldBeSkipped` I can add a simple `get` for each list, and call it in the `usvfs` class with a function that does what `..ShouldBeSkipped` essentially already does, but just takes the list from the previous `get` call.

I'll toy on this more when I find some time, very nice catch!

Reply:
PR is ready for review again